org.eclipse.emfcloud.modelserver.client/src/main/java/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java

```diff
@@ -151,45 +154,64 @@ public void close() {
    }
 
    @Override
-   public CompletableFuture<Response<String>> update(final String modelUri, final String updatedModel) {
-      final Request request = new Request.Builder()
-         .url(
-            createHttpUrlBuilder(makeUrl(MODEL_BASE_PATH))
-               .addQueryParameter("modeluri", modelUri)
-               .build())
-         .patch(
-            RequestBody.create(
-               Json.object(Json.prop("data", Json.text(updatedModel))).toString(), MediaType.parse("application/json")))
-         .build();
+   public CompletableFuture<Response<String>> create(final String modelUri, final String createdModelAsJsonText) {
+      TextNode dataNode = Json.text(createdModelAsJsonText);
+      final Request request = buildCreateOrUpdateRequest(modelUri, POST, "json", dataNode);
 
       return makeCall(request)
-         .thenApply(response -> parseField(response, "data"))
-         .thenApply(this::getBodyOrThrow);
+              .thenApply(response -> parseField(response, "data"))
+              .thenApply(this::getBodyOrThrow);
```

Comment:
I commented 2 problems in the checkstyle file (see my other comment below). 
Then I was able to run the checks, but there were no complains from checkstyle. The 2 lines above are not mentioned as violating the code style.

I can modify the 2 lines above, that is no problem, but the system to prevent this is not working, as checkstyle does not report issues/warmings.

Reply:
The formatting preference is currently defined via Eclipse settings. When using another IDE (like Intelli) this is not working and the checkstyle configuration does not validate the formatting style.

I open #24 to track this. 
For now please modify this two lines manually.

Reply:
Sorry, my fault. Found out that it is the code formatting rules. 
I will apply them.